### PR TITLE
Add Red Hat Integration - Camel K operator

### PR DESCRIPTION
**Pull request by pierDipi**
This PR adds Red Hat Integration - Camel K operator

Product docs: https://access.redhat.com/documentation/en-us/red_hat_integration/2023.q2/html/getting_started_with_camel_k/index

I've tested it with:

```shell
$ oc apply -k https://github.com/pierDipi/gitops-catalog/red-hat-camel-k/operator/overlays/1.10.x?ref=add-camel-k
```

on OCP 4.13.

Once the installation is complete I get the default integrationplatform instance:
```shell
$ oc get integrationplatforms.camel.apache.org -A
NAMESPACE         NAME      PHASE
red-hat-camel-k   camel-k   Ready
```

and a Kamelet

```yaml
apiVersion: camel.apache.org/v1alpha1
kind: KameletBinding
metadata:
  name: timer-source-pipe
spec:
  source:
    ref:
      kind: Kamelet
      apiVersion: camel.apache.org/v1alpha1
      name: timer-source
    properties:
      message: hello timer-source Kamelet
  sink:
    properties:
      cloudEventsType: com.redhat.timer.source
    ref:
      kind: Broker
      apiVersion: eventing.knative.dev/v1
      name: broker
```

that delivers events like this one:

```
☁️  cloudevents.Event
Context Attributes,
  specversion: 1.0
  type: com.redhat.timer.source
  source: source
  id: B90C0C9949D9227-0000000000000245
  time: 2023-12-12T14:17:24.327Z
  datacontenttype: text/plain
Data,
  hello timer-source Kamelet
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by strangiato — APPROVED** [review]
lgtm

**Review by strangiato — CHANGES_REQUESTED** [review]
Can you add the missing new line the linter flagged?

looks good besides that.

Thank you for the contribution!

File: red-hat-camel-k/instance/integration-platform/README.md
Comment (christophd) [inline red-hat-camel-k/instance/integration-platform/README.md:3]: I think the IntegrationPlatform comes with the operator installation automatically. Do we really need this custom IntegrationPlatform resource?

**Review by christophd — COMMENTED** [review]
Looks good as far as I understand how the redhat-cop works.

The only thing I would suggest to review is the need for the custom IntegrationPlatform resource. This should be created automatically with the Camel K operator installation (in the operator namespace) and should be valid for all integrations being managed by the operator.

File: red-hat-camel-k/INFO.md
Comment (pierDipi) [reply red-hat-camel-k/INFO.md:22]: This documentation was generated by the Operator metadata / docs itself, perhaps we should improve that first ?

File: red-hat-camel-k/INFO.md
Comment (pierDipi) [reply red-hat-camel-k/INFO.md:34]: Same as the other comment, this documentation was generated by the Operator metadata / docs itself, perhaps we should improve that first ?

Another option is to remove `INFO.md` completely, so that we don't enter into the details here as it's not the best place (and it will probably get outdated eventually)

File: red-hat-camel-k/instance/integration-platform/README.md
Comment (pierDipi) [reply red-hat-camel-k/instance/integration-platform/README.md:3]: Ok, I'll remove the instance directory

File: red-hat-camel-k/INFO.md
Comment (pierDipi) [reply red-hat-camel-k/INFO.md:34]: @christophd any thoughts here?

Comment (christophd) [reply red-hat-camel-k/INFO.md:34]: @pierDipi ah ok, agreed thanks! in case this is generated from the Operator metadata this PR is not the right place to fix but the Operator metadata itself. We can leave `INFO.md` as it is and update as soon as the Operator metadata has improved. It is not a big issue and the information given is not wrong. I was just wondering whether YAML would be a better fit here

**Comment by pierDipi** [comment]
@strangiato @christophd please review this again

**Review by strangiato — APPROVED** [review]
lgtm

**Comment by strangiato** [comment]
@christophd I will defer to you for your expertise on the Camel operator before merging.

**Comment by christophd** [comment]
LGTM, many thanks! 👍 The comments that I have had are resolved because the changes need to be done directly in the operator metadata.

In case we update and improve the operator metadata on our side is this going to be automatically updated here?

**Review by christophd — APPROVED** [review]

**Pull request merged**
